worker/worker.go

```diff
@@ -90,15 +106,45 @@ func (w *worker) punishPlayers(ctx context.Context) {
 }
 
 func (w *worker) punishPlayer(ctx context.Context, id string) {
+	var playerName string
+	var grid string
+	// Fetch player info to get name and grid position
 	err := w.pool.WithConnection(ctx, func(c *rconv2.Connection) error {
+		players, err := c.Players(ctx)
+		if err != nil {
+			return err
+		}
+		for _, p := range players.Players {
+			if p.Id == id && p.Position.IsSpawned() {
+				playerName = p.Name
+				grid = p.Position.Grid(w.current).String()
+				break
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		w.l.Error("fetch-player-for-punish", "player_id", id, "error", err)
+		return
+	}
+
+	// Punish the player
+	err = w.pool.WithConnection(ctx, func(c *rconv2.Connection) error {
```

Comment:
This is rather inefficient. If we want to keep the playername and grid info thing, then both usages of a connection should be merged. No need to grab a connection from the pool and then return it just to more or less instantly grab another one.